core-data/resolvers.js

```diff
@@ -28,3 +33,14 @@ export async function* getMedia( state, id ) {
 	const media = await apiRequest( { path: `/wp/v2/media/${ id }` } );
 	yield receiveMedia( media );
 }
+
+/**
+ * Requests a post type element from the REST API.
+ *
+ * @param {Object} state State tree
+ * @param {number} slug  Post Type slug
+ */
+export async function* getPostType( state, slug ) {
+	const postType = await apiRequest( { path: `/wp/v2/types/${ slug }?context=edit` } );
```

Comment:
The context is hard-coded for now. As it's the only context we use right now.
I wonder how should we approach this:

 - Always use `edit` assuming this context returns the full object?
 - Store received data per context and make the context and argument of the resolver (I don't like this)
 - Something else (maybe server-side)?

What's your take on this?

Reply:
It's a quite interesting question. It looks like `view` is the default one: https://developer.wordpress.org/rest-api/reference/post-types/#arguments. Given that this can be also customized by the user we should come up with a general solution which allows to provide this value. I have a more general question, why we have `edit` here and `view` in other places? Shouldn't we use the same context for all requests on a given page?

Reply:
We use `edit` mainly because it contains more information. For me it's, this argument is a bit ambiguous because it assumes that it has to do with rights but rights shouldn't be asked for using request arguments. The server could just reply with all the fields the user can see (unless we ask for specific fields using the fields arg)

Reply:
My naive assumption is that `context` should be defined on the server when loading a given page and automatically attached to all requests. This would enforce consistency and provide a unified way of handling all endpoints.

Reply:
Tough question. Another option is to make selectors more granular, so that we know based on the selector being called whether we need to provide `context=edit` (i.e. Why do we need it specifically for what we're refactoring? Can those needs be expressed as distinct selectors?). Of course, this means we run into other issues of how to avoid duplicate requests amongst all of the selectors' resolvers.

A main annoyance for me with this context business is that it's authorized as though the user intends to edit specific values, but in many cases it's just generally useful information (like labels of a post type) which doesn't seem to be such sensitive information, at least to be blocking users of certain roles from accessing.

Reply:
Ok! let's move forward with the current implementation (forcing `edit`) and see how it plays out with other data.